api/v1alpha1/nodeobservability_types.go

```diff
@@ -29,8 +29,8 @@ const (
 
 // NodeObservabilitySpec defines the desired state of NodeObservability
 type NodeObservabilitySpec struct {
-	// Labels is map of key:value pairs that are used to match against node labels
-	Labels map[string]string `json:"labels,omitempty"`
+	// NodeSelector is map of key:value pairs that are used to match against node labels
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
```

Comment:
Having empty `NodeSelector` doesn't have much sense anymore as we don't update all the workers.
I recommend to set it as `+kubebuilder:validation:Required` to enforce the user to specify some nodeselector.